### PR TITLE
Exploit P10 instructions in existing and new quad shift operations.

### PR DESCRIPTION
**Pull request by munroesj52**
Use P10 instructions for P10 specific implementations of existing
shift left double quadword and new shift left/right by bits immediate
operations.

Found that the new vrlq, vslq, vsraw, vsrq instructions use vector
bits 57:63 for shift count where original Altivec Octet and Bit shifts
used bits 121:127.  So P10 needs shift count adjusted (xxsplatd)
for P10.  Also updated vec_sldqi to leverage vsldbi for P10.

Add associated compile and unit tests.

	* src/pveclib/vec_int128_ppc.h (vec_rlq [_ARCH_PWR10]):
	Use vec_splatd() to adjust shift count for P10.
	* (vec_sldqi): Add code to special case Octet, (shb < 8),
	and (shb > 7) cases for P10 using vsldbi.
	(vec_slq [_ARCH_PWR10]): Use vec_splatd() to adjust shift
	count for P10.
	(vec_sraq [_ARCH_PWR10]): Use vec_splatd() to adjust shift
	count for P10.
	(vec_srq [_ARCH_PWR10]): Use vec_splatd() to adjust shift
	count for P10.

	* src/testsuite/arith128_test_i128.c (test_vsldqi): Add tests
	for boundary conditions specific to new implementation and P10.

	* src/testsuite/vec_int128_dummy.c (test_vec_sldqi_7):
	New compiler test.

	* src/testsuite/vec_pwr10_dummy.c (test_setb_sq_PWR10,
	test_vec_sldqi_7_PWR10, test_vec_sldqi_15_PWR10,
	test_vec_sldqi_16_PWR10, test_vec_sldqi_127_PWR10):
	New compiler tests.

Signed-off-by: Steven Munroe <munroesj52@gmail.com>

### CODE REVIEW

File: src/pveclib/vec_int128_ppc.h [hunk 1]
```diff
@@ -5815,22 +5817,42 @@ vec_sldq (vui128_t vrw, vui128_t vrx, vui128_t vrb)
...
+	    // Shift left double quad (256-bits) by Octet
+            h = vec_sld ((vui8_t) vrw, (vui8_t) vrx, (shb / 8));
+            l = vec_sld ((vui8_t) vrx, (vui8_t) vrx, (shb / 8));
+            // Then Shift Left Double by Bit to complete the shift.
+  	    result = vec_vsldbi ((vui128_t) h, (vui128_t) l, (shb % 8));
+#else       // Load shb as vector and use general vec_sldq case.
+	    const vui8_t vrb = vec_splats ((unsigned char) shb);
+	    result = vec_sldq (vrw, vrx, (vui128_t) vrb);
+#endif
+	  }
```
Comment (ThinkOpenly) [inline src/pveclib/vec_int128_ppc.h:5850]: indentation is wrong on a couple levels here

File: src/pveclib/vec_int128_ppc.h [hunk 2]
```diff
@@ -5815,22 +5817,42 @@ vec_sldq (vui128_t vrw, vui128_t vrx, vui128_t vrb)
...
     }
   else
     {
       const vui8_t vrb = vec_splats ((unsigned char) shb);
-      result = (vui8_t) vec_sldq (vrw, vrx, (vui128_t) vrb);
+      result = vec_sldq (vrw, vrx, (vui128_t) vrb);
     }
 
   return ((vui128_t) result);
```
Comment (ThinkOpenly) [inline src/pveclib/vec_int128_ppc.h:5858]: You don't need this cast anymore, with the retyping of `result` to `vui128_t`.

**Review by ThinkOpenly — APPROVED** [review]
just a couple of nits, otherwise :+1:

File: src/pveclib/vec_int128_ppc.h (hunk 1 above)
Comment (munroesj52) [reply src/pveclib/vec_int128_ppc.h]: spaces vs tabs. Convert tabs to spaces,

File: src/pveclib/vec_int128_ppc.h (hunk 2 above)
Comment (munroesj52) [reply src/pveclib/vec_int128_ppc.h:5858]: Ok
